src/cpp/transport/udp/UDPServerWindows.cpp

```diff
@@ -21,17 +21,23 @@ namespace uxr {
 UDPServer::UDPServer(uint16_t port, uint16_t discovery_port)
     : UDPServerBase(port),
       poll_fd_{},
-      buffer_{0}
-{
-    (void) discovery_port;
-}
+      buffer_{0},
+      discovery_server_(*processor_, port_, discovery_port)
+{}
 
 bool UDPServer::init(bool discovery_enabled)
 {
-    (void) discovery_enabled;
-
     bool rv = false;
 
+    /* Init discovery. */
+    if (discovery_enabled)
+    {
+        if (!discovery_server_.run())
+        {
+            return false;
```

Comment:
Are we using the one return function style in C++ too?

Reply:
Yes, I think so.

Reply:
This function has two return statements, then.

Reply:
The style allows using a return at the beginning of the function for preconditions.